metron-platform/metron-pcap-backend/src/test/java/org/apache/metron/pcap/query/PcapCliTest.java

```diff
@@ -112,7 +120,24 @@ public void runs_fixed_pcap_filter_job_with_default_argument_list() throws Excep
     return new TypeSafeMatcher<Map<K, V>>() {
       @Override
       protected boolean matchesSafely(Map<K, V> item) {
-        return item.entrySet().containsAll(map.entrySet());
+        for(K key: map.keySet()) {
+          if (key.equals(PcapOptions.HADOOP_CONF.getKey())) {
+            Configuration itemConfiguration = (Configuration) item.get(PcapOptions.HADOOP_CONF.getKey());
+            Map<String, Object> mapConfiguration = (Map<String, Object>) map.get(PcapOptions.HADOOP_CONF.getKey());
+            for(String setting: mapConfiguration.keySet()) {
+              if (!mapConfiguration.get(setting).equals(itemConfiguration.get(setting, ""))) {
+                return false;
+              }
+            }
+          } else {
+            V itemValue = item.get(key);
+            V mapValue = map.get(key);
+            if (itemValue != null ? !itemValue.equals(mapValue) : mapValue == null) {
```

Comment:
Should this be `!(mapValue == null)`?

**Similarly stated:**
```
if (itemValue == null ? mapValue == null : itemValue.equals(mapValue)) {
  return true;
}
```

If both values are null, it should return `true`, I would expect.

Reply:
If any key/value pair doesn't match, they we want to return false.  If all key/value pairs match, then we return true.  With your change, true will be returned the first time a key/value pair matches and all others are skipped.

Reply:
@merrimanr, sorry I provided a bad example - I was just trying to rewrite the conditional in a different way with less "nots", rather than suggest that exact change. More precisely, this looks correct to me:
```
 if (itemValue != null ? !itemValue.equals(mapValue) : mapValue != null) {
...
```
as this will treat instances where BOTH values are `null` as equal.

Reply:
Fixed